azurelinuxagent/ga/env.py

```diff
@@ -133,6 +134,10 @@ def _operation(self):
 
         success = self._osutil.enable_firewall(dst_ip=self._protocol.get_endpoint(), uid=os.getuid())
 
+        if self._log_initial_firewall_list:
```

Comment:
Do we really need this anymore? The only time where `enable_firewall` rule will not run on service start is if the rules are already present, and in that case, do we need to list the rules? Isin't it enough to just list them whenever the `enable_firewall` actually sets the rules?

Reply:
Actually, two cases one is service start and second if network setup service already enabled the firewall (Happy path scenario all the rules present all the time). In those cases, if we don't log these, while debugging livesites it's hard to know what rules already setup.

Reply:
I would like to send telemetry whenever the `enable_firewall` actually sets the rules. One is to exercise new changes and second is to understand how frequently we are readding these rules and also, figure out how bad the containers hitting missing rules and etc.

Reply:
> if we don't log these, while debugging livesites it's hard to know what rules already setup.

If that is the goal, then wont it make more sense to log it periodically rather than just doing it once on service start?

> I would like to send telemetry whenever the enable_firewall actually sets the rules.

Correct me if I'm wrong, but I dont think you're doing that right now in the current code, all I see is logging, no telemetry. And speaking of which, could you add it there lol.

> One is to exercise new changes and second is to understand how frequently we are readding these rules 

Totally with you on the 2nd one but not sure I follow you on the first one, can you elaborate a bit more please

Reply:
`If that is the goal, then wont it make more sense to log it periodically rather than just doing it once on service start?`

I feel adding periodically is too much, you end of seeing this in entire log. In general, on agent start you would expect to see firewall rules setup. So, in this you would never see in log after restart. I feel that's really confuses.

` Correct me if I'm wrong, but I dont think you're doing that right now in the current code, all I see is logging, no telemetry. And speaking of which, could you add it there lol.`
I'm not. I'm gonna add now.

`Totally with you on the 2nd one but not sure I follow you on the first one, can you elaborate a bit more please`
I mean to check the code path hitting if we get telemetry.

Reply:
> I feel adding periodically is too much, you end of seeing this in entire log.

Depends on the periodicity you set no? Say you set it to once a day or something, then it wont be as bad. 

> In general, on agent start you would expect to see firewall rules setup

I mean in a general case, either the `env` thread will set it or the `network-setup.service` will right? In both the cases we have enough logging to tell us when they do. I think the case that you're trying to capture is if by some weird chance we see a scenario where the rules are not set or messed around with then we should have some sort of log. Is that understanding wrong? If that indeed is the case, wont a periodic log for it make more sense? We'll have a periodic check to make sure nobody is messing up our rules, and try to capture the state of it if we do detect a change.

 > So, in this you would never see in log after restart. 

What restart are you talking about? The service or the machine? Because in both cases we would see the extra log right?

Reply:
`env thread will set it or the network-setup.service will right?`

Yes, we log if `env `thread detect change but not on `network-setup.service`. Since this happens on boot, and we don't have access to agent logs at that time so don't log anything. Obliviously now env thread won't detect any change if network service already did setup.

Reply:
we do log it, if the agent detects the service was setup, it fetches the logs from the setup service. We dont log the firewall rules but we log that the firewall rules were added properly. 

This discussion is dragging on longer than it needs to. Essentially all I'm trying to say is that if your goal is to log the firewall rules to check the state of those rules because you don't trust our services setup the rules properly or if there was some external intervention, we should do it periodically than just once on service start (periodically will also capture the first time on service start). Periodically will capture more debug state and would be more useful long term. (Additionally implementing this would be trivial too, just add a new PeriodicOperation on the env thread for logging the rules)

I personally don't understand the need for logging it once on service start especially since in your changes you're checking the availability of each of those rules and if any of them is missing adding them back and subsequently logging it and I think that's enough.

That's my 2 cents on it, I'll let you decide which route you want to take on this.

Reply:
If you go by env thread detect change for logging, in most of the time you won't hit that scenario. I'm down to period logging once a day to avoid too much logging and also, gives enough debug data overtime. Thanks for the suggestion.